doc/configuration.rst

```diff
@@ -605,13 +605,13 @@ NetworkRKUSBLoader
 ~~~~~~~~~~~~~~~~~~~
 A NetworkRKUSBLoader describes an `RKUSBLoader`_ available on a remote computer.
 
-AndroidFastboot
-~~~~~~~~~~~~~~~
-An AndroidFastboot resource describes a USB device in the fastboot state.
+AndroidUsbFastboot
```

Comment:
I am not really happy with the resource name. It should be "USB" (to keep things consistent with the other resource names). Is there a reason you added "Usb" in the middle?

For the time being we're stuck with:
- AndroidFastboot
- NetworkAndroidFastboot
- AndroidNetFastboot
- AndroidUsbFastboot

Pretty confusing..

Reply:
Hi Bastian,

thx for the review.

> It should be "USB" (to keep things consistent with the other resource names)

Fine for me. We wanted to keep it consistent with the new resource `AndroidNetFastboot`. So we should change this also to `AndroidNETFastboot`?


 
> For the time being we're stuck with:
> 
> * AndroidFastboot
> * NetworkAndroidFastboot
> * AndroidNetFastboot
> * AndroidUsbFastboot
> 
> Pretty confusing..

Our intention was to make it clear that the user knows wich resource to use by only reading the name of the resource in the documentation. But if the maintainers are fine to keep the current resource name `AndroidFastboot` for the USB backend resource, its fine for me too. The benifit is, that no deprecation warning is needed anymore and also no breaking changes for users in the future when die resource `AndroidFastboot` would be removed.

Reply:
I'd like to take this chance to clean up a bit. We have and keep for compatibility:
- AndroidFastboot
- NetworkAndroidFastboot

New names for the above:
- AndroidUSBFastboot
- RemoteAndroidUSBFastboot

And for the new network functionality:
- AndroidNetFastboot
- RemoteAndroidNetFastboot

Reply:
> New names for the above:
> 
> * AndroidUSBFastboot
> * RemoteAndroidUSBFastboot

By testing the changed PR with the above renaming we get another question we need to discuss here. `AndroidFastboot` and `AndroidUSBFastboot` are exported by the same class `USBGenericExport` in [exporter.py](https://github.com/labgrid-project/labgrid/blob/dffc61d31db002d674ce2a31cff9795760d65682/labgrid/remote/exporter.py#L494). But this class sets the `Network` prefix. So the new resource name `RemoteAndroidUSBFastboot` is not handled here. Changing the class `USBGenericExport` would result in changing all `Network*` resources handled by this class. Creating a new class that handles the `Remote` prefix is not the  most elegant way but would allow us the renaming? (see my updated changes)? What do you think?

Reply:
Yes, it's not elegant, but at least it's explicit. :) I think we'll have to live with this.